app/controllers/teams_controller.rb

```diff
@@ -39,12 +40,28 @@ def update
     change_description(@team, :team)
   end
 
+  def typeahead
+    authorize @team
+    search = []
+    valid_users = User.enabled.where.not(id: @team.team_users.pluck(:user_id))
```

Comment:
Could you move this logic into the User model ? That is, I think that the User model should have a `search_from_query` method (or one with a better name :P) with this logic inside of it. Then, provide unit tests for this specific method. I'm saying this because I'm not really comfortable with queries that are not trivial in a controller.